### PR TITLE
Introduce StormTopologyBuilder

### PR DESCRIPTION
**Pull request by ttim**
This PR introduces `StormTopologyBuilder` class and move all `Topology` creation code from `Storm` platform to it, to make it easier to read.

Apart from that this PR doesn't contain any changes in logic.

### CODE REVIEW

File: summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import com.twitter.summingbird.storm.Constants._
+import com.twitter.summingbird.storm.option.AnchorTuples
+import com.twitter.summingbird.storm.planner.StormNode
+import org.apache.storm.generated.StormTopology
+import org.apache.storm.topology.TopologyBuilder
+import org.apache.storm.{ Config => BacktypeStormConfig }
+import org.slf4j.LoggerFactory
+import scala.reflect.ClassTag
+
+case class StormTopologyBuilder(options: Map[String, Options], jobId: JobId, stormDag: Dag[Storm]) {
```
Comment (pankajroark) [inline summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala]: Doc comment.

File: summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  }
+
+  private def scheduleSpout(node: SourceNode[Storm], topologyBuilder: TopologyBuilder) = {
+    val nodeName = stormDag.getNodeName(node)
+    val (sourceParalleism, stormSpout) = SpoutProvider(this, node).apply
+    topologyBuilder.setSpout(nodeName, stormSpout, sourceParalleism)
+  }
+
+  private def scheduleSummerBolt(node: SummerNode[Storm], topologyBuilder: TopologyBuilder): Unit = {
+
```
Comment (pankajroark) [inline summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala]: remove extra line

File: summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      val maxEmitPerExecute = getOrElse(node, DEFAULT_MAX_EMIT_PER_EXECUTE)
+      logger.info(s"[$nodeName] maxEmitPerExecute : ${maxEmitPerExecute.get}")
+
+      val maxExecutePerSec = getOrElse(node, DEFAULT_MAX_EXECUTE_PER_SEC)
+      logger.info(s"[$nodeName] maxExecutePerSec : $maxExecutePerSec")
+
+      val storeBaseFMOp = { op: (ExecutorKeyType, (Option[ExecutorValueType], ExecutorValueType)) =>
+        val ((k, batchID), (optiVWithTS, (ts, v))) = op
+        val optiV = optiVWithTS.map(_._2)
```
Comment (pankajroark) [inline summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala]: I think we use opt as the prefix everywhere else so may be use that here as val. i.e optVWithTS and optV.

**Review by pankajroark — APPROVED** [review]
Nice refactoring, thanks.

**Review by ulyssepence — APPROVED** [review]
OK looks like an OK change. This is just to encapsulate some of the build logic?

File: summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala (hunk 1 above)
Comment (ttim) [reply summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala]: Done

File: summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala (hunk 2 above)
Comment (ttim) [reply summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala]: Done

File: summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala (hunk 3 above)
Comment (ttim) [reply summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormTopologyBuilder.scala]: Done

**Comment by ttim** [comment]
@benpence yes, just to encapsulate topology build logic into one place.

**Comment by pankajroark** [comment]
lgtm

**Pull request merged**